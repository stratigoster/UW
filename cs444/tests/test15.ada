PACKAGE foo IS
  type Week is (Sun,Mon,Tue,Wed,Thu,Fri,Sat);
  type WeekEnd is (Sat,Sun);
  myFaveDay : Week := Sun; -- overloading works!
  b : BOOLEAN := Sun > Mon OR ELSE Tue = myFaveDay;
  x : BOOLEAN := Sun > Sat;
--               ^^^^^^^^^
-- ambiguous operands for comparison (since both Sun and Sat are either
-- of type Week or WeekEnd

  y : BOOLEAN := Sun > Mon AND Sat /= Tue; -- this is OK
END foo;
