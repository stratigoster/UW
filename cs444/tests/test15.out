***ERROR***:6:22:	Ambiguous operands for comparison

------
Compilation error
------
